pkg/repository/backend_postgres.go

```
@@ -1367,3 +1377,134 @@ func (r *PostgresBackendRepository) UpdateSecret(ctx context.Context, workspace
+		return nil, err
+	}
+
+	var s types.ScheduledJob
+	query = `
+	INSERT INTO scheduled_job (job_id, job_name, job_schedule, job_payload, deployment_id, stub_id)
+	VALUES ($1, $2, $3, $4::jsonb, $5, $6)
+	RETURNING id, external_id, job_id, job_schedule, job_payload, stub_id, created_at, updated_at, deleted_at;
+	`
+	if err := r.client.GetContext(ctx, &s, query, jobId, scheduledJob.JobName, scheduledJob.Schedule, payloadJSON, scheduledJob.DeploymentId, scheduledJob.StubId); err != nil {
+		r.unscheduleCron(ctx, *jobId)
+		return nil, err
+	}
+
+	return &s, nil
+}
+
+func (r *PostgresBackendRepository) GetScheduledJob(ctx context.Context, deploymentId uint) (*types.ScheduledJob, error) {
+	var scheduledJob types.ScheduledJob
+	query := `
+	SELECT sj.*
+	FROM scheduled_job sj
+	JOIN cron.job j ON sj.job_id = j.jobid
+	JOIN deployment d ON sj.deployment_id = d.id
+	WHERE sj.deployment_id = $1 AND sj.deleted_at IS NULL;
+	`
+	err := r.client.GetContext(ctx, &scheduledJob, query, deploymentId)
+	if err != nil {
+		return nil, err
+	}
+
+	return &scheduledJob, nil
+}
+
+func (r *PostgresBackendRepository) DeleteScheduledJob(ctx context.Context, scheduledJob *types.ScheduledJob) error {
+	updateQuery := `
+	UPDATE scheduled_job
+	SET deleted_at = CURRENT_TIMESTAMP
+	WHERE id = $1;
+	`
+	if _, err := r.client.ExecContext(ctx, updateQuery, scheduledJob.Id); err != nil {
+		return fmt.Errorf("failed to delete scheduled job: %v", err)
+	}
+
+	if err := r.unscheduleCron(ctx, scheduledJob.JobId); err != nil {
+		return fmt.Errorf("failed to unschedule job: %v", err)
+	}
+
+	return nil
+}
+
+func (r *PostgresBackendRepository) unscheduleCron(ctx context.Context, jobId uint64) error {
+	unscheduleQuery := `SELECT cron.unschedule($1::integer);`
+	if _, err := r.client.ExecContext(ctx, unscheduleQuery, jobId); err != nil {
+		return fmt.Errorf("failed to call cron.unschedule function: %v", err)
```

can we just return the err directly?

---

Updated.